### PR TITLE
Add principals into security appender

### PR DESCRIPTION
**Pull request by lucasponce**
@jshaughn: I've started to add the discussed changes for https://github.com/kiali/kiali/issues/2924 into the security appender.

I think these changes make sense but perhaps I'm less familiar with the structure.

Feel free to comment.

Note: it's not yet finished, tests and other checks are missing.

### CODE REVIEW

File: graph/config/cytoscape/cytoscape.go [hunk 1]
```diff
@@ -88,9 +88,11 @@ type EdgeData struct {
...
 
 	// App Fields (not required by Cytoscape)
-	Traffic      ProtocolTraffic `json:"traffic,omitempty"`      // traffic rates for the edge protocol
-	ResponseTime string          `json:"responseTime,omitempty"` // in millis
-	IsMTLS       string          `json:"isMTLS,omitempty"`       // set to the percentage of traffic using a mutual TLS connection
+	Traffic      	ProtocolTraffic `json:"traffic,omitempty"`      // traffic rates for the edge protocol
+	ResponseTime 	string          `json:"responseTime,omitempty"` // in millis
+	IsMTLS       	string          `json:"isMTLS,omitempty"`       // set to the percentage of traffic using a mutual TLS connection
+	DestPrincipal 	string		 	`json:"destPrincipal,omitempty"`
+	SourcePrincipal string       	`json:"sourcePrincipal,omitempty"`
```
Comment (jshaughn) [inline graph/config/cytoscape/cytoscape.go:95]: Just needs format a short comments for each new field.  For bonus points alphabetize below // App Fields.

File: graph/config/cytoscape/cytoscape.go [hunk 2]
```diff
@@ -282,6 +284,12 @@ func buildConfig(trafficMap graph.TrafficMap, nodes *[]*NodeWrapper, edges *[]*E
 					Protocol: protocol,
 				},
 			}
+			if e.Metadata[graph.SourcePrincipal] != nil {
+				ed.SourcePrincipal = e.Metadata[graph.SourcePrincipal].(string)
+			}
+			if e.Metadata[graph.DestPrincipal] != nil {
+				ed.DestPrincipal = e.Metadata[graph.DestPrincipal].(string)
+			}
```
Comment (jshaughn) [inline graph/config/cytoscape/cytoscape.go:292]: Alphabetize (just flip them)

File: graph/meta.go [hunk 3]
```diff
@@ -30,6 +30,8 @@ const (
 	IsUnused        MetadataKey = "isUnused"
 	ProtocolKey     MetadataKey = "protocol"
 	ResponseTime    MetadataKey = "responseTime"
+	SourcePrincipal MetadataKey = "sourcePrincipal"
+	DestPrincipal   MetadataKey = "destPrincipal"
```
Comment (jshaughn) [inline graph/meta.go:34]: Alphabetize

File: graph/telemetry/istio/appender/security_policy.go [hunk 4]
```diff
@@ -90,6 +90,12 @@ func (a SecurityPolicyAppender) appendGraph(trafficMap graph.TrafficMap, namespa
 	a.populateSecurityPolicyMap(securityPolicyMap, &inVector)
 
 	applySecurityPolicy(trafficMap, securityPolicyMap)
+
+	principalMap := make(map[string]map[graph.MetadataKey]string)
+	a.populatePrincipalMap(principalMap, &outVector)
+	a.populatePrincipalMap(principalMap, &inVector)
```
Comment (jshaughn) [inline graph/telemetry/istio/appender/security_policy.go:96]: Not sure you really need to do all of this separately, I'd suggest maybe just working this into the existing 'populate' and 'apply' and not have to walk through everything twice.  If you are worried about getting the feature in then this is fine for now and I'll probably refactor it later.

**Review by jshaughn — CHANGES_REQUESTED** [review]
I think this is the right way to do it, piggybacking on the security appender is nice.  I think it's perfectly reasonable to require the 'Security' Display option.  Note that opetions can be automatically enabled if and when that makes sense.  For example, if you type 'has mtls' into graph find you'll see that the security option gets enabled automagically.

A few minor comments about alphabetizing, a more major comment about removing the redundancy, but the main blocker is that you need to add to the test code.  If you need help let me know, you could progress with the main feature and when I have time I can offer a sub-pr.

**Comment by lucasponce** [comment]
Thanks @jshaughn, let me know how it looks like now.

**Review by jshaughn — APPROVED** [review]
Seems fine, maybe we can optimize a little more but certainly looks good enough to progress.

**Pull request merged**